lean.py

```diff
@@ -62,9 +62,9 @@ def score_func(sentence):
 
 
 def filterLean(msg):
-    m = re.findall('```([Ll]ean)?(.*?)```', msg, re.MULTILINE|re.DOTALL)
+    m = re.findall('```([Ll]ean4?)?(.*?)```', msg, re.MULTILINE|re.DOTALL)
     r = "\n".join([x[1] for x in m])
-    r = r.replace('\n#eval', '\n--#eval') # skip evaluations
+    # r = r.replace('\n#eval', '\n--#eval') # skip evaluations
```

Comment:
You removed this, but I don't see an alternative fix for it?

Reply:
catching pexpect.exceptions.EOF should handle this alright

Reply:
But it does make the check very slow if there's an infinite loop...